### ISSUE TITLE
[Bug]: Unable to use pako in create-react-app 5 with latest babel v7.20.x

### ISSUE THREAD

**Issue by cmdcolin** [babel/babel#15132]
### 💻

- [ ] Would you like to work on a fix?

### How are you using Babel?

Other (Next.js, Gatsby, vue-cli, ...)

### Input code

The file from pako is node_modules/pako/dist/pako.esm.mjs   

```
const gen_bitlen = (s, desc) =>
//    hello_deflate_state *s;
//    tree_desc *desc;    /* the tree descriptor */
{
  const tree            = desc.dyn_tree;
  const max_code        = desc.max_code;
 
```

The error from a build is

```
$ yarn build
yarn run v1.22.18
$ react-scripts build
Creating an optimized production build...
Failed to compile.

Module parse failed: Unexpected token (255:129)
File was processed with these loaders:
 * ./node_modules/babel-loader/lib/index.js
 * ./node_modules/source-map-loader/dist/cjs.js
You may need an additional loader to handle the result of these loaders.
|  *     not null.
|  */
> var gen_bitlen = function gen_bitlen(s, desc) /*    hello_deflate_state *s;*/ /*    tree_desc *desc;    /* the tree descriptor */*/{
|   var tree = desc.dyn_tree;
|   var max_code = desc.max_code;


```

### Configuration file name

package.json

### Configuration

I use a default version of installing create-react-app with yarn.lock in https://github.com/cmdcolin/pako_error_in_cra5

Can eject the cra config to see details

### Current and expected behavior

This I believe is a fairly recently change, and I'm not certain it's babel that does this type of conversion of comment formats. If this is not true I can redirect this issue elsewhere!

The code worked when my yarn.lock has e.g. ~7.19.x versions of babel packages, but started failing now at ~7.20.x. there are many babel packages so I'm not sure which but a diff between working and non-working is at this pastebin https://pastebin.com/xe3HSAgW

### Environment

- babel 7.20.x (failing) babel 7.19.x (working)
- node 18.20
- yarn 1.22.18
- ubuntu 22.10
- not a monorepo but testing in a create react app

reproducible repo here https://github.com/cmdcolin/pako_error_in_cra5

**Comment by cmdcolin** [comment]
Might be dupe of https://github.com/babel/babel/issues/15111 can close if so!

**Comment by cmdcolin** [comment]
i'll just close my own issue, it's very likely a dupe :)

**Comment by JLHwung** [comment]
It seems like a new issue.

The line comment `//    tree_desc *desc;    /* the tree descriptor */` is valid, however Babel generates a block comment `/* tree_desc /* tree descriptor */ */` which completely messes up with the `/* */` pair within the line comment.

**Comment by JLHwung** [comment]
@cmdcolin The fix of #15111 has been published in 7.20.2, can you reproduce this issue on 7.20.2?

**Comment by cmdcolin** [comment]
I think https://github.com/cmdcolin/pako_error_in_cra5 is using 7.20.2 of @babel/core

can see in yarn.lock for details

and it does still produce the error. do you think this issue should re-open?

**Comment by srijanreddy98** [comment]
This is happening to me in angular as well.
`./node_modules/pako/lib/zlib/trees.js:257:106 - Error: Module parse failed: Unexpected token (257:106)
File was processed with these loaders:
 * ./node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js
 * ./node_modules/@angular-devkit/build-angular/node_modules/@ngtools/webpack/src/ivy/index.js
You may need an additional loader to handle the result of these loaders.
|  *     not null.
|  */
> function gen_bitlen(s, desc) /*    deflate_state *s;*/ /*    tree_desc *desc;    /* the tree descriptor */*/{
|   var tree = desc.dyn_tree;
|   var max_code = desc.max_code;`

@JLHwung Any eta on when the fix will be published?

**Comment by JBCodeWorld** [comment]
Same error with React JS.

**Comment by liuxingbaoyu** [comment]
@nakulkundra https://github.com/angular/angular-cli/blob/main/packages/angular_devkit/build_angular/package.json#L13-L14
It seems that `angular` is still using the old version, can you try https://docs.npmjs.com/cli/v8/configuring-npm/package-json#overrides?

**Comment by nakulkundra** [comment]
@liuxingbaoyu  tried what you suggested , but no luck,.

```
 "overrides": {
    "@angular-devkit/build-angular": {
      "@babel/generator": "7.20.3"
    }}
```

Not working, getting the below error. [](url)


```
./node_modules/@progress/kendo-drawing/node_modules/pako/dist/pako_deflate.js:2504:112 - Error: Module parse failed: Unexpected token (2504:112)
File was processed with these loaders:
 * ./node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js
You may need an additional loader to handle the result of these loaders.
|        *     not null.
|        */
>       function gen_bitlen(s, desc) /*    deflate_state *s;*/ /*    tree_desc *desc;    /* the tree descriptor */*/{
|         var tree = desc.dyn_tree;
|         var max_code = desc.max_code;

./node_modules/pdfmake/build/pdfmake.js:43324:116 - Error: Module parse failed: Unexpected token (43324:116)
File was processed with these loaders:
 * ./node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js
You may need an additional loader to handle the result of these loaders.
|            *     not null.
|            */
>           function gen_bitlen(s, desc) /*    deflate_state *s;*/ /*    tree_desc *desc;    /* the tree descriptor */*/{
|             var tree = desc.dyn_tree;
|             var max_code = desc.max_code;

./node_modules/xlsx/jszip.js:7853:112 - Error: Module parse failed: Unexpected token (7853:112)
File was processed with these loaders:
 * ./node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js
You may need an additional loader to handle the result of these loaders.
|        *     not null.
|        */
>       function gen_bitlen(s, desc) /*    deflate_state *s;*/ /*    tree_desc *desc;    /* the tree descriptor */*/{
|         var tree = desc.dyn_tree;
|         var max_code = desc.max_code;
```


Package File 


```
{
  "_from": "@angular-devkit/build-angular@13.3.9",
  "_id": "@angular-devkit/build-angular@13.3.9",
  "_inBundle": false,
  "_integrity": "sha512-1LqcMizeabx3yOkx3tptCSAoEhG6nO6hPgI/B3EJ07G/ZcoxunMWSeN3P3zT10dZMEHhcxl+8cSStSXaXj9hfA==",
  "_location": "/@angular-devkit/build-angular",
  "_phantomChildren": {
    "@babel/code-frame": "7.18.6",
    "@babel/helper-compilation-targets": "7.20.0",
    "@babel/helper-module-transforms": "7.20.2",
    "@babel/helpers": "7.20.1",
    "@babel/parser": "7.20.3",
    "@babel/traverse": "7.20.1",
    "@babel/types": "7.20.2",
    "@jridgewell/source-map": "0.3.2",
    "@types/eslint-scope": "3.7.4",
    "@types/estree": "0.0.51",
    "@webassemblyjs/ast": "1.11.1",
    "@webassemblyjs/wasm-edit": "1.11.1",
    "@webassemblyjs/wasm-parser": "1.11.1",
    "acorn-import-assertions": "1.8.0",
    "brace-expansion": "1.1.11",
    "browserslist": "4.21.4",
    "chrome-trace-event": "1.0.3",
    "convert-source-map": "1.9.0",
    "debug": "4.3.4",
    "enhanced-resolve": "5.10.0",
    "es-module-lexer": "0.9.3",
    "esbuild-android-arm64": "0.14.22",
    "esbuild-darwin-64": "0.14.22",
    "esbuild-darwin-arm64": "0.14.22",
    "esbuild-freebsd-64": "0.14.22",
    "esbuild-freebsd-arm64": "0.14.22",
    "esbuild-linux-32": "0.14.22",
    "esbuild-linux-64": "0.14.22",
    "esbuild-linux-arm": "0.14.22",
    "esbuild-linux-arm64": "0.14.22",
    "esbuild-linux-mips64le": "0.14.22",
    "esbuild-linux-ppc64le": "0.14.22",
    "esbuild-linux-riscv64": "0.14.22",
    "esbuild-linux-s390x": "0.14.22",
    "esbuild-netbsd-64": "0.14.22",
    "esbuild-openbsd-64": "0.14.22",
    "esbuild-sunos-64": "0.14.22",
    "esbuild-windows-32": "0.14.22",
    "esbuild-windows-64": "0.14.22",
    "esbuild-windows-arm64": "0.14.22",
    "escape-string-regexp": "1.0.5",
    "eslint-scope": "5.1.1",
    "events": "3.3.0",
    "gensync": "1.0.0-beta.2",
    "glob-to-regexp": "0.4.1",
    "graceful-fs": "4.2.10",
    "jsesc": "2.5.2",
    "json-parse-better-errors": "1.0.2",
    "json5": "2.2.1",
    "loader-runner": "4.3.0",
    "lodash": "4.17.21",
    "lru-cache": "6.0.0",
    "mime-types": "2.1.35",
    "neo-async": "2.6.2",
    "ora": "5.4.1",
    "run-async": "2.4.1",
    "safer-buffer": "2.1.2",
    "schema-utils": "3.1.1",
    "signal-exit": "3.0.7",
    "source-map-support": "0.5.21",
    "string-width": "4.2.3",
    "strip-ansi": "6.0.1",
    "tapable": "2.2.1",
    "terser-webpack-plugin": "5.3.6",
    "through": "2.3.8",
    "tmp": "0.0.33",
    "type-fest": "0.21.3",
    "watchpack": "2.4.0",
    "webpack-sources": "3.2.3"
  },
  "_requested": {
    "type": "version",
    "registry": true,
    "raw": "@angular-devkit/build-angular@13.3.9",
    "name": "@angular-devkit/build-angular",
    "escapedName": "@angular-devkit%2fbuild-angular",
    "scope": "@angular-devkit",
    "rawSpec": "13.3.9",
    "saveSpec": null,
    "fetchSpec": "13.3.9"
  },
  "_requiredBy": [
    "#DEV:/"
  ],
  "_resolved": "https://registry.npmjs.org/@angular-devkit/build-angular/-/build-angular-13.3.9.tgz",
  "_shasum": "400cc6e7c1662c66929077c511b2aa0eab76d9be",
  "_spec": "@angular-devkit/build-angular@13.3.9",
  "_where": "C:\\Code",
  "author": {
    "name": "Angular Authors"
  },
  "bugs": {
    "url": "https://github.com/angular/angular-cli/issues"
  },
  "builders": "builders.json",
  "bundleDependencies": false,
  "dependencies": {
    "@ampproject/remapping": "2.2.0",
    "@angular-devkit/architect": "0.1303.9",
    "@angular-devkit/build-webpack": "0.1303.9",
    "@angular-devkit/core": "13.3.9",
    "@babel/core": "7.16.12",
    "@babel/generator": "7.16.8",
    "@babel/helper-annotate-as-pure": "7.16.7",
    "@babel/plugin-proposal-async-generator-functions": "7.16.8",
    "@babel/plugin-transform-async-to-generator": "7.16.8",
    "@babel/plugin-transform-runtime": "7.16.10",
    "@babel/preset-env": "7.16.11",
    "@babel/runtime": "7.16.7",
    "@babel/template": "7.16.7",
    "@discoveryjs/json-ext": "0.5.6",
    "@ngtools/webpack": "13.3.9",
    "ansi-colors": "4.1.1",
    "babel-loader": "8.2.5",
    "babel-plugin-istanbul": "6.1.1",
    "browserslist": "^4.9.1",
    "cacache": "15.3.0",
    "circular-dependency-plugin": "5.2.2",
    "copy-webpack-plugin": "10.2.1",
    "core-js": "3.20.3",
    "critters": "0.0.16",
    "css-loader": "6.5.1",
    "esbuild": "0.14.22",
    "esbuild-wasm": "0.14.22",
    "glob": "7.2.0",
    "https-proxy-agent": "5.0.0",
    "inquirer": "8.2.0",
    "jsonc-parser": "3.0.0",
    "karma-source-map-support": "1.4.0",
    "less": "4.1.2",
    "less-loader": "10.2.0",
    "license-webpack-plugin": "4.0.2",
    "loader-utils": "3.2.0",
    "mini-css-extract-plugin": "2.5.3",
    "minimatch": "3.0.5",
    "open": "8.4.0",
    "ora": "5.4.1",
    "parse5-html-rewriting-stream": "6.0.1",
    "piscina": "3.2.0",
    "postcss": "8.4.5",
    "postcss-import": "14.0.2",
    "postcss-loader": "6.2.1",
    "postcss-preset-env": "7.2.3",
    "regenerator-runtime": "0.13.9",
    "resolve-url-loader": "5.0.0",
    "rxjs": "6.6.7",
    "sass": "1.49.9",
    "sass-loader": "12.4.0",
    "semver": "7.3.5",
    "source-map-loader": "3.0.1",
    "source-map-support": "0.5.21",
    "stylus": "0.56.0",
    "stylus-loader": "6.2.0",
    "terser": "5.14.2",
    "text-table": "0.2.0",
    "tree-kill": "1.2.2",
    "tslib": "2.3.1",
    "webpack": "5.70.0",
    "webpack-dev-middleware": "5.3.0",
    "webpack-dev-server": "4.7.3",
    "webpack-merge": "5.8.0",
    "webpack-subresource-integrity": "5.1.0"
  },
  "deprecated": false,
  "description": "Angular Webpack Build Facade",
  "engines": {
    "node": "^12.20.0 || ^14.15.0 || >=16.10.0",
    "npm": "^6.11.0 || ^7.5.6 || >=8.0.0",
    "yarn": ">= 1.13.0"
  },
  "homepage": "https://github.com/angular/angular-cli",
  "keywords": [
    "angular",
    "Angular CLI",
    "devkit",
    "sdk",
    "Angular DevKit"
  ],
  "license": "MIT",
  "main": "src/index.js",
  "name": "@angular-devkit/build-angular",
  "optionalDependencies": {
    "esbuild": "0.14.22"
  },
  "peerDependencies": {
    "@angular/compiler-cli": "^13.0.0 || ^13.3.0-rc.0",
    "@angular/localize": "^13.0.0 || ^13.3.0-rc.0",
    "@angular/service-worker": "^13.0.0 || ^13.3.0-rc.0",
    "karma": "^6.3.0",
    "ng-packagr": "^13.0.0",
    "protractor": "^7.0.0",
    "tailwindcss": "^2.0.0 || ^3.0.0",
    "typescript": ">=4.4.3 <4.7"
  },
  "peerDependenciesMeta": {
    "@angular/localize": {
      "optional": true
    },
    "@angular/service-worker": {
      "optional": true
    },
    "karma": {
      "optional": true
    },
    "ng-packagr": {
      "optional": true
    },
    "protractor": {
      "optional": true
    },
    "tailwindcss": {
      "optional": true
    }
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/angular/angular-cli.git"
  },
  "typings": "src/index.d.ts",
  "version": "13.3.9"
}

```
`

**Comment by JLHwung** [comment]
@nakulkundra What's the `npm version`? The `overrides` is supported in 8.3+.

**Comment by srijanreddy98** [comment]
Hi @JLHwung. I did added the ovveride:
```
  "overrides": {
    "@angular-devkit/build-angular": {
      "@babel/generator": "7.20.3"
    }
  }
```

My npm version: 8.19.3. But I am still getting the error.

 I was checking through the package lock. I found two instances of babel generator:

```"node_modules/@babel/generator": {
      "version": "7.20.3",
      "resolved": "https://registry.npmjs.org/@babel/generator/-/generator-7.20.3.tgz",
      "integrity": "sha512-Wl5ilw2UD1+ZYprHVprxHZJCFeBWlzZYOovE4SDYLZnqCOD11j+0QzNeEWKLLTWM7nixrZEh7vNIyb76MyJg3A==",
      "dependencies": {
        "@babel/types": "^7.20.2",
        "@jridgewell/gen-mapping": "^0.3.2",
        "jsesc": "^2.5.1"
      },
      "engines": {
        "node": ">=6.9.0"
      }
    },
```

and 

```
    "@babel/generator": {
      "version": "7.20.3",
      "resolved": "https://registry.npmjs.org/@babel/generator/-/generator-7.20.3.tgz",
      "integrity": "sha512-Wl5ilw2UD1+ZYprHVprxHZJCFeBWlzZYOovE4SDYLZnqCOD11j+0QzNeEWKLLTWM7nixrZEh7vNIyb76MyJg3A==",
      "requires": {
        "@babel/types": "^7.20.2",
        "@jridgewell/gen-mapping": "^0.3.2",
        "jsesc": "^2.5.1"
      },
      "dependencies": {
        "@jridgewell/gen-mapping": {
          "version": "0.3.2",
          "resolved": "https://registry.npmjs.org/@jridgewell/gen-mapping/-/gen-mapping-0.3.2.tgz",
          "integrity": "sha512-mh65xKQAzI6iBcFzwv28KVWSmCkdRBWoOh+bYQGW3+6OZvbbN3TqMGo5hqYxQniRcH9F2VZIoJCm4pa3BPDK/A==",
          "requires": {
            "@jridgewell/set-array": "^1.0.1",
            "@jridgewell/sourcemap-codec": "^1.4.10",
            "@jridgewell/trace-mapping": "^0.3.9"
          }
        }
      }
    },
```

Any ideas?

**Comment by JLHwung** [comment]
@srijanreddy98 What does `npm ls @babel/generator` print? You can also wait for https://github.com/angular/angular-cli/pull/24193

**Comment by srijanreddy98** [comment]
@JLHwung it prints:
oppia@3.2.8 /Users/.../Documents/oppia/oppia
├─┬ @angular-devkit/build-angular@13.3.9
│ ├─┬ @angular/localize@13.3.11
│ │ └─┬ @babel/core@7.17.2
│ │   └── @babel/generator@7.20.3 deduped
│ ├─┬ @babel/core@7.16.12
│ │ └── @babel/generator@7.20.3 deduped
│ └── @babel/generator@7.20.3
├─┬ @angular/compiler-cli@13.3.11
│ └─┬ @babel/core@7.20.2
│   └── @babel/generator@7.20.3 deduped
└─┬ babel-eslint@10.1.0
  └─┬ @babel/traverse@7.20.1
    └── @babel/generator@7.20.3 deduped
```

**Comment by srijanreddy98** [comment]
@JLHwung really appreciate your help. I was not able to fix it on the babel side of things. It was proving very difficult to pin down versions on babel side with no results popping up in my favour. (We typically use yarn but I even tried npm which has an easier way with overrides than yarn resolutions).


### For folks coming to this issue looking for resolutions:
I resolved by pinning down the version of Pako to 2.1.0 (the latest version just released at the time of the comment to fix the babel problems). Check npm override or yarn resolutions to pin it down. To figure out which packages you need to pin the pako version: run `npm ls pako` to get a dependency tree. Non ideal solution but give it a try.

My `npm ls pako` returned:
```
oppia@3.2.8 /Users/redvasa/Documents/oppia/oppia
├─┬ @angular-devkit/build-angular@13.3.9
│ └─┬ piscina@3.2.0
│   └─┬ hdr-histogram-js@2.0.3
│     └── pako@^1.0.3
└─┬ browserify-zlib@0.2.0
  └── pako@~1.0.5
```

So I added (for yarn):
```
  "resolutions": {
    "@angular-devkit/build-angular/piscina/hdr-histogram-js/pako": "2.1.0",
    "browserify-zlib/pako": "2.1.0"
  }
```
